agent/component/core.go

```diff
@@ -33,35 +33,21 @@ const (
 // It is recommended that all bubbly agent components embed ComponentCore.
 type ComponentCore struct {
 	Type ComponentType
-	// NATSServer contains the connections to the NATS Server.
-	// This is nested within the ComponentCore as Components are
-	// individually responsible for managing their own connection
-	// to a NATS Server.
-	NATSServer *NATS
+	// EConn contains the an encoded connection to the NATS server
+	EConn *nats.EncodedConn
```

Comment:
A comment on this: my choice for establishing an encoded connection at time of publish/request (i.e., in a `Component`'s `Publish` and `Request` methods) is because the `EncodedConn` seemed to care about the underlying connection encoding, which might differ. Actually, we currently only ever use the `nats.DEFAULT_ENCODER`, and therefore your re-design will not break anything. However, from my understanding of this rework, any future desire to change this may be trickier, as now we're opting for a component-scoped encoded connection rather than a publication/request-scoped one.